delta/plugins/storage/src/main/scala/ch/epfl/bluebrain/nexus/delta/plugins/storage/StoragesConfig.scala

```diff
@@ -0,0 +1,116 @@
+package ch.epfl.bluebrain.nexus.delta.plugins.storage
+
+import akka.http.scaladsl.model.Uri
+import ch.epfl.bluebrain.nexus.delta.sdk.syntax._
+import ch.epfl.bluebrain.nexus.delta.plugins.config.{AggregateConfig, IndexingConfig}
+import ch.epfl.bluebrain.nexus.delta.plugins.storage.StoragesConfig.StorageTypeConfig
+import ch.epfl.bluebrain.nexus.delta.plugins.storage.model.DigestAlgorithm
+import ch.epfl.bluebrain.nexus.delta.sdk.cache.KeyValueStoreConfig
+import ch.epfl.bluebrain.nexus.delta.sdk.model.identities.AuthToken
+import ch.epfl.bluebrain.nexus.delta.sdk.model.permissions.Permission
+import ch.epfl.bluebrain.nexus.delta.sdk.model.search.PaginationConfig
+
+import java.nio.file.Path
+
+/**
+  * Configuration for the Storages module.
+  *
+  * @param aggregate         configuration of the underlying aggregate
+  * @param keyValueStore     configuration of the underlying key/value store
+  * @param pagination        configuration for how pagination should behave in listing operations
+  * @param indexing          configuration of the indexing process
+  * @param storageTypeConfig configuration of each of the storage types
+  */
+final case class StoragesConfig(
+    aggregate: AggregateConfig,
+    keyValueStore: KeyValueStoreConfig,
+    pagination: PaginationConfig,
+    indexing: IndexingConfig,
+    storageTypeConfig: StorageTypeConfig
+)
+
+object StoragesConfig {
+
+  /**
+    * The configuration of each of the storage types
+    * @param disk          configuration for the disk storage
+    * @param amazon        configuration for the s3 compatible storage
+    * @param remoteDisk    configuration for the remote disk storage
+    */
+  @SuppressWarnings(Array("OptionGet"))
+  final case class StorageTypeConfig(
+      disk: DiskStorageConfig,
+      amazon: Option[S3StorageConfig],
+      remoteDisk: Option[RemoteDiskStorageConfig]
+  ) {
+    lazy val amazonUnsafe: S3StorageConfig             = amazon.get
+    lazy val remoteDiskUnsafe: RemoteDiskStorageConfig = remoteDisk.get
+  }
+
+  /**
+    * Disk storage configuration
+    *
+    * @param volume          the base [[Path]] where the files are stored
+    * @param digestAlgorithm algorithm for checksum calculation
+    * @param readPermission  the default permission required in order to download a file from a disk storage
+    * @param writePermission the default permission required in order to upload a file to a disk storage
+    * @param showLocation    flag to decide whether or not to show the absolute location of the files in the metadata response
+    * @param maxFileSize     the default maximum allowed file size (in bytes) for uploaded files
+    */
+  final case class DiskStorageConfig(
+      volume: Path,
+      digestAlgorithm: DigestAlgorithm,
+      readPermission: Permission,
```

Comment:
the last 4 ones could be included in a common subsetting as they are common to every storage.
And if we do the same thing with the equivalent values in StorageValue, it should make it easy combine them where needed

Reply:
It is easier for deriving the pureconfig like this, and I don't have any other reason why to group it

Reply:
It is easier for deriving the pureconfig like this, and I don't have any other reason why to group it